### PR TITLE
Add translate_sid processor to Winlogbeat

### LINKED ISSUE #7451: [Winlogbeat] - Security Group Management MemberSid translation
Need a way to enable sid translation for MemberSid on Security Group Management events. 

Example: event_id: 4732

When an object (User or Group) is added to a group, only the SID of object is available in most events. There are two fields, MemberName and MemberSid, MemberName is usually '-' and should trigger a SID lookup based on the event code. Local SID lookup for local Security Group Management events, and domain lookup for every thing else.

According to this link, it's possible that only local group events are impacted by the MemberName always being set to '-':
https://www.ultimatewindowssecurity.com/securitylog/encyclopedia/event.aspx?eventID=4732

### PR DESCRIPTION
**Pull request by andrewkroh**
The `translate_sid` processor translates a Windows security identifier (SID)
into an account name. It retrieves the name of the account associated with the
SID, the first domain on which the SID is found, and the type of account.

Closes #7451

### CODE REVIEW

File: libbeat/docs/processors-list.asciidoc [hunk 1]
```diff
@@ -95,6 +95,9 @@ endif::[]
 ifndef::no_truncate_fields_processor[]
 * <<truncate-fields, `truncate_fields`>>
 endif::[]
+ifdef::include_translate_sid_processor[]
+* <<processor-translate-sid, `translate_sid`>>
+endif::[]
```
Comment (urso) [inline libbeat/docs/processors-list.asciidoc]: @dedemorton We switched to docs/beats excluding what they don't support. Shall we be more consistent and use `ifndef::no_translate_sid_processor`?

**Comment by urso** [comment]
Metricbeat and Filebeat allow for generic events to be pushed to them. E.g. `winlogbeat->file->filebeat`. Would it make sense to include the processor with these two as well? How about packetbeat correlating events with processes?

File: libbeat/docs/processors-list.asciidoc (hunk 1 above)
Comment (dedemorton) [reply libbeat/docs/processors-list.asciidoc]: yes let's be consistent.

Comment (andrewkroh) [reply libbeat/docs/processors-list.asciidoc]: Done.

**Comment by andrewkroh** [comment]
@urso I moved the processor to libbeat so any Beat can use it now.

**Pull request merged**